src/tools/chdman.cpp

```diff
@@ -2485,6 +2482,42 @@ static void do_extract_cd(parameters_map &params)
 		{
 			output_toc_file->printf("%d\n", toc.numtrks);
 		}
+		else if (mode == MODE_NORMAL)
+		{
+			bool mode1 = false;
+			bool mode2 = false;
+			bool cdda = false;
+
+			for (int tracknum = 0; tracknum < toc.numtrks; tracknum++)
+			{
+				switch (toc.tracks[tracknum].trktype)
+				{
+					case cdrom_file::CD_TRACK_MODE1:
+					case cdrom_file::CD_TRACK_MODE1_RAW:
+						mode1 = true;
+						break;
+
+					case cdrom_file::CD_TRACK_MODE2:
+					case cdrom_file::CD_TRACK_MODE2_FORM1:
+					case cdrom_file::CD_TRACK_MODE2_FORM2:
+					case cdrom_file::CD_TRACK_MODE2_FORM_MIX:
+					case cdrom_file::CD_TRACK_MODE2_RAW:
+						mode2 = true;
+						break;
+
+					case cdrom_file::CD_TRACK_AUDIO:
+						cdda = true;
+						break;
+				}
+			}
+
+			if (cdda && !mode1 && !mode2)
+				output_toc_file->printf("CD_DA\n\n\n");
+			else if (mode2)
+				output_toc_file->printf("CD_ROM_XA\n\n\n");
+			else
+				output_toc_file->printf("CD_ROM\n\n\n");
```

Comment:
Wouldn’t this be simpler as:
```
			if (mode2)
				output_toc_file->printf("CD_ROM_XA\n\n\n");
			else if (cdda && !mode1)
				output_toc_file->printf("CD_DA\n\n\n");
			else
				output_toc_file->printf("CD_ROM\n\n\n");
```

It’s always `CD_ROM_XA` if Mode 2 tracks are present, so that only needs to be tested once.

Reply:
You're right. I updated the code.